### PR TITLE
Vpc route module

### PR DESCRIPTION
**Pull request by Polina-Gubina**
Depends-On: https://github.com/opentelekomcloud/ansible-collection-cloud/pull/30

### CODE REVIEW

**Comment by Polina-Gubina** [comment]
recheck

**Comment by Polina-Gubina** [comment]
Depends-On: opentelekomcloud/ansible-collection-cloud#{{30}}

**Comment by Polina-Gubina** [comment]
recheck

**Comment by Polina-Gubina** [comment]
recheck

**Comment by Polina-Gubina** [comment]
recheck

**Comment by Polina-Gubina** [comment]
Depends-On: #30

**Comment by Polina-Gubina** [comment]
recheck

**Comment by Polina-Gubina** [comment]
recheck

**Comment by Polina-Gubina** [comment]
recheck

**Review by anton-sidelnikov — DISMISSED** [review]

**Comment by tischrei** [comment]
check for conflicts.

File: plugins/modules/vpc_route.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DOCUMENTATION = '''
+---
+module: vpc_route
+short_description: Add/Delete vpc route from OpenTelekomCloud
+extends_documentation_fragment: opentelekomcloud.cloud.otc
+version_added: "0.0.3"
```
Comment (tischrei) [inline plugins/modules/vpc_route.py:19]: version 0.2.0

File: tests/integration/targets/vpc_route/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      register: dropped
+
+    - name: assert result
+      assert:
+        that:
+          - dropped is success
+          - dropped is changed
+
+  always:
+    - block:
```
Comment (tischrei) [inline tests/integration/targets/vpc_route/tasks/main.yaml:125]: ignore_errors: yes is missing

File: plugins/modules/vpc_route.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    project_id:
+        description: Project id.
+        returned: On success when C(state=present)
+        type: str
+        sample: "6ysa5bac-0925-4d5b-add8-cb6667b8"
+'''
+
+EXAMPLES = '''
+# Create a vpc route.
+- vpc_route:
```
Comment (tischrei) [inline plugins/modules/vpc_route.py:88]: pls provide full qualified cname opentelekomcloud.cloud.modulname

File: tests/integration/targets/vpc_route/tasks/main.yaml [hunk 4]
```diff
@@ -0,0 +1,160 @@
+---
+- module_defaults:
+    vpc_peering:
```
Comment (tischrei) [inline tests/integration/targets/vpc_route/tasks/main.yaml:3]: complete module name is missing, so it fails for me.

File: tests/integration/targets/vpc_route/tasks/main.yaml [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        state: present
+        network: admin_external_net
+        enable_snat: True
+        interfaces:
+          - net: "{{ test_network_2.network.name }}"
+            subnet: "{{ test_subnet_2.subnet.name }}"
+      register: test_router_2
+
+    - name: Create vpc peering
+      vpc_peering:
```
Comment (tischrei) [inline tests/integration/targets/vpc_route/tasks/main.yaml:77]: the same as above

File: tests/integration/targets/vpc_route/tasks/main.yaml [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+        name: "{{ vpc_peering_name }}"
+        local_router: "{{ test_router_1.router.name }}"
+        project_id_local: "{{ project_id }}"
+        peer_router: "{{ test_router_2.router.name }}"
+        project_id_peer: "{{ project_id }}"
+      register: vpc_peering
+
+    - name: Create vpc route -check mode
+      vpc_route:
```
Comment (tischrei) [inline tests/integration/targets/vpc_route/tasks/main.yaml:86]: module name

File: tests/integration/targets/vpc_route/tasks/main.yaml [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+      register: vpc_route_check
+      check_mode: true
+
+    - name: assert result
+      assert:
+        that:
+          - vpc_route_check is changed
+
+    - name: Create vpc route
+      vpc_route:
```
Comment (tischrei) [inline tests/integration/targets/vpc_route/tasks/main.yaml:99]: module name

File: tests/integration/targets/vpc_route/tasks/main.yaml [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+      register: vpc_route
+
+    - name: assert result
+      assert:
+        that:
+          - vpc_route is success
+          - vpc_route is changed
+
+    - name: Drop existing vpc route
+      vpc_route:
```
Comment (tischrei) [inline tests/integration/targets/vpc_route/tasks/main.yaml:112]: module name

File: tests/integration/targets/vpc_route/tasks/main.yaml [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+      assert:
+        that:
+          - dropped is success
+          - dropped is changed
+
+  always:
+    - block:
+      # Cleanup
+      - name: Drop vpc peering
+        vpc_peering:
```
Comment (tischrei) [inline tests/integration/targets/vpc_route/tasks/main.yaml:128]: module name

**Review by tischrei — APPROVED** [review]

**Review by tischrei — CHANGES_REQUESTED** [review]
Please correct the comments

File: plugins/modules/vpc_route.py [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+    router_id:
+        description: The router of the route.
+        returned: On success when C(state=present)
+        type: str
+        sample: "4dae5bac-0725-2d5b-add8-cb6667b8"
+    project_id:
+        description: Project id.
+        returned: On success when C(state=present)
+        type: str
+        sample: "6ysa5bac-0925-4d5b-add8-cb6667b8"
```
Comment (anton-sidelnikov) [inline plugins/modules/vpc_route.py:83]: use 2 instead 4 spaces

File: plugins/modules/vpc_route.py [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+    def _check_route(self, destination, router_id):
+
+        query = {}
+        result = True
+
+        query['destination'] = destination
+        query['vpc_id'] = router_id
+        data = []
+
+        for raw in self.conn.vpc.routes(**query):
```
Comment (anton-sidelnikov) [inline plugins/modules/vpc_route.py:129]: if raw null?

**Comment by Polina-Gubina** [comment]
recheck

File: tests/integration/targets/vpc_route/tasks/main.yaml [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+        assert:
+          that:
+            - dropped is success
+            - dropped is changed
+
+      - name: Drop vpc peering
+        opentelekomcloud.cloud.vpc_peering:
+          name: "{{ vpc_peering_name }}"
+          state: absent
+        ignore_errors: yes
```
Comment (anton-sidelnikov) [inline tests/integration/targets/vpc_route/tasks/main.yaml:132]: you can remove that

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]
look at int test

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Review by gtema — APPROVED** [review]

**Review by tischrei — APPROVED** [review]

**Pull request merged**